snuba/datasets/dataset.py

```diff
@@ -150,15 +171,16 @@ def __init__(
             query_plan_builder=query_plan_builder,
             abstract_column_set=abstract_column_set,
             writable_storage=writable_storage,
+            column_resolver=column_resolver,
         )
         # Convenience columns that evaluate to a bucketed time. The bucketing
         # depends on the granularity parameter.
-        # The bucketed time column names cannot be overlapping with existing
-        # schema columns
+        # The bucketed time column names must reference columns defined in the logical
+        # schema
```

Comment:
Now that we have a logical schema separate from a physical one (thus the logical schema is not used to build clickhouse tables anymore) I don't see any reason not to make time bucketed columns part of the logical schema. They are columns that can be referenced in queries after all.

Reply:
This makes a lot of sense to me.